src/lib/components/button/main-button.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  const getButtonState = (disabled?: boolean) => {
+    if (disabled) {
+      return 'disabled';
+    }
+    return 'default';
+  };
+
+  const getButtonStyle = (variant) => {
```

Comment:
Is this any different than just returning the string directly?

We could probably just enforce this with TypeScript:

```ts
const getButtonStyle = (variant: 'primary' | 'secondary | 'destroy') => variant;
```

But, even then, there is probably a simpler way express this that I'll stumble across later on in the code.